ykrt/src/compile/jitc_yk/codegen/x86_64/mod.rs

```diff
@@ -1095,10 +1100,10 @@ impl<'a> X64CodeGen<'a> {
                 }
             }
             LocalAlloc::Register => todo!(),
-            LocalAlloc::ConstInt(c) => {
-                // FIXME: Adjust for different constant sizes. Requires storing the size in the
-                // variant.
-                dynasm!(self.asm; mov Rq(reg.code()), QWORD *c as i64);
+            LocalAlloc::ConstInt { bits, v } => {
+                // FIXME: Adjust for different constant sizes?
```

Comment:
I know you didn't introduce this, but I think this FIXME can be removed. If the type is smaller than 64-bits then the top bits are just undefined and two's compliment ensures any truncation maintains the same numeric value.